source/braille.py

```diff
@@ -1224,15 +1250,18 @@ def focus(self, region):
 			self.windowEndPos = end
 
 	def update(self):
+		self.rawText = ""
 		self.brailleCells = []
 		self.cursorPos = None
 		start = 0
 		if log.isEnabledFor(log.IO):
 			logRegions = []
 		for region in self.visibleRegions:
+			rawText = region.rawText
 			if log.isEnabledFor(log.IO):
-				logRegions.append(region.rawText)
+				logRegions.append(rawText)
 			cells = region.brailleCells
+			self.rawText+=rawText

```

Comment:
concatinating a string like this is costly, especially when done many times. Strings are imutable so Python keeps creating a new one. It would be better to store them in a list, and do a join at the end.

Reply:
Never mind. Timing tests seem to show this is probably optimized out anyway.  Not too sure how exactly, but if we're fully not sure, it is better to leave it as is.